### PR TITLE
[UNTESTED] [WEB EDIT] Adds a prequel reference

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

If a target of the Tesla Blast spell is holding an esword, esaber, a desword, or a null rod that looks like an esword (or esaber), they can reflect the electricty of the spell back at the caster, disfiguring them.

## Why It's Good For The Game

Epic prequel (and original trilogy, and new trilogy (how the fuck do you get beaten 3 times by the same trick, Palp?!)) reference.

I wanted to add an achievement for doing this, but I don't wanna mess with icon code.

## Changelog
:cl: ATHATH
add: Trying to use the Tesla Blast spell against an esword wielder causes something special to happen.
/:cl:

### CODE REVIEW

File: code/modules/spells/spell_types/lightning.dm [hunk 1]
```diff
@@ -65,22 +65,47 @@
 /obj/effect/proc_holder/spell/targeted/tesla/proc/Bolt(mob/origin,mob/target,bolt_energy,bounces,mob/user = usr)
 	origin.Beam(target,icon_state="lightning[rand(1,12)]",time=5)
 	var/mob/living/carbon/current = target
-	if(current.anti_magic_check())
+	var/whywouldyoueverusethisagainstajedi = FALSE
```
Comment (tralezab) [inline code/modules/spells/spell_types/lightning.dm]: this is outrageous, its unfair

File: code/modules/spells/spell_types/lightning.dm [hunk 2]
```diff
@@ -65,22 +65,47 @@
...
+	var/helditem=current.get_active_held_item()
+	if(helditem)
+		if(istype(helditem, /obj/item/melee/transforming/energy/sword) && helditem.active) //only works if they have an esword in their active hand
+			whywouldyoueverusethisagainstajedi = TRUE
+		else if(istype(helditem, /obj/item/twohanded/dualsaber) && helditem.wielded) //or a double-bladed esword
+			whywouldyoueverusethisagainstajedi = TRUE
+		else if(istype(helditem, /obj/item/nullrod/claymore/saber)) //or one of the null rod variants that look like eswords
+			whywouldyoueverusethisagainstajedi = TRUE
```
Comment (tralezab) [inline code/modules/spells/spell_types/lightning.dm]: all of this is bad and can be done in a much better way (first thought is a typecached list)

**Review by tralezab — CHANGES_REQUESTED** [review]
basically this pr is verified af

**Comment by nemvar** [comment]
unlimited POWAHH

File: code/modules/spells/spell_types/lightning.dm (hunk 1 above)
Comment (trollbreeder) [reply code/modules/spells/spell_types/lightning.dm]: One, you're removing the magic check for some reason.
Two, that var name is ridiculous.
I'm thinking esword_counter would be good.

Comment (ATH1909) [reply code/modules/spells/spell_types/lightning.dm]: @trollbreeder nah, I just moved the antimagic check to further down

File: code/modules/spells/spell_types/lightning.dm (hunk 2 above)
Comment (trollbreeder) [reply code/modules/spells/spell_types/lightning.dm]: Dont forget your spaces as well to make the code look cleaner.

File: code/modules/spells/spell_types/lightning.dm [hunk 3]
```diff
@@ -65,22 +65,51 @@
...
+		if(!whywouldyoueverusethisagainstajedi) //if the caster is also holding a lightsaber, they can ping pong the electricity back and forth with a jedi until it dissipates
+			if(current.electrocute_act(bolt_energy,"Lightning Bolt",flags = SHOCK_NOGLOVES) && current == user) //you got your lightning reflected back at you, and you're not immune to it
+				current.emote("scream")
+				ADD_TRAIT(current, TRAIT_DISFIGURED, "got palp'd")
+				return     //alright, meme's over, guys, let's not bounce to anyone else
 		var/list/possible_targets = new
 		for(var/mob/living/M in view(range,target))
+			if(user == M && current!=user && whywouldyoueverusethisagainstajedi && los_check(current,M))
+				current.visible_message("<span class='warning'>[current]'s [helditem] reflects the spell's electricity, sending it back to [M]!</span>", "<span class='userdanger'>Your [helditem] reflects the spell's electricity, sending it back to [M]!</span>") //even if the jedi is, like, the third guy in the chain, I'm still gonna use the word "reflect", because the electricity is still being sent back to the caster
+				Bolt(current,M,max((bolt_energy-5),5),bounces-1,user)
```
Comment (trollbreeder) [inline code/modules/spells/spell_types/lightning.dm]: All your space are belong to us.
```suggestion
				Bolt(current, M, max((bolt_energy - 5), 5), bounces-1, user)
```

**Comment by optimumtact** [comment]
untested webedit

**Comment by trollbreeder** [comment]
Clean up your code damnit.

**Pull request closed without merging**